publiq-ui/pub-modal.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <b-modal
+    cancel-variant="outline-secondary"
+    :title="title"
+    :ok-title="confirmTitle"
+    :cancel-title="cancelTitle"
+    title-class="title"
+    modal-class="modal"
+    content-class="content"
+    body-class="body"
+    :visible="visible"
+    :scrollable="false"
+    :hide-footer="isContentVariant"
+    :hide-header="isQuestionVariant"
+    :size="isContentVariant ? 'xl' : 'md'"
```

Comment:
```
:hide-footer="isContentVariant"
:hide-header="isQuestionVariant"
:size="isContentVariant ? 'xl' : 'md'"
```

With all of these differences, I'd consider having 2 separate components like `pub-modal` and `pub-confirmation-modal` or something like that.

Also because the following props are not relevant for the content variant:
```
:ok-title="confirmTitle"
:cancel-title="cancelTitle"
```

There's no clear threshold when it's best to split up, but my gut feeling says these modals are quite distinct.

When we also manually add the listeners as I suggested previously, we'll also have listeners that are not relevant for one variant or the other.

Reply:
They're both modals though, if these two are already too different for them to be variants, won't we want to split up every variant of components? except maybe the buttons and the alerts, since their only difference is the color.
I get that there are some differences in the props inside pub-modal, but visually I think they are the same component with slight differences.

Reply:
_Personally_ I'd draw the line when the difference is not just styling/colors. Once we start displaying/hiding certain parts and have props that only apply to one specific variant, it usually quickly results in a lot of `if/else` switches and confusing documentation. For example, I wouldn't be surprised if at some point we have to introduce an `alert` variant for modals, which only shows the `ok` button but not the cancel button. Then you need to add some more checks to show the footer but only the ok button etc and the code will become very unclear with all of these conditions in one component I believe. But curious what @geoffreydhuyvetters or @brampauwelyn think of this.

Reply:
IMO I think the modal component should only contain the black overlay + the dialog box + event to close the modal.
I wouldn't add too much markup (such as a title) in the component itself. 
In this case you're always free to add all the elements which are necessary for a specific case.

Reply:
Under normal circumstances I'd agree, but sadly we (have to) use Bootstrap components underneath right now to replicate the look and feel of the AngularJS app in the Vue app, and the Bootstrap modal component is quite bloated in that aspect.

> \<b-modal\>, by default, has an OK and Cancel buttons in the footer. These buttons can be customized by setting various props on the component. You can customize the size of the buttons, disable buttons, hide the Cancel button (i.e. ok-only), choose a variant (e.g. danger for a red OK button) using the ok-variant and cancel-variant props, and provide custom button content using the ok-title and cancel-title props, or using the named slots modal-ok and modal-cancel.

https://bootstrap-vue.org/docs/components/modal

So it seems hard to get a "blank" modal unless we build it ourselves with Bootstrap classes to replicate the look of the AngularJS app.

Reply:
@geoffreydhuyvetters we (Simon, Bram, I) discussed this in a call and after weighing the options the consencus was that multiple components would probably be best in this case. So `pub-content-modal`, `pub-question-modal`, ... (for example, exact naming wasn't discussed).

Only downside is that the two variants right now share the same styling, so we'd need to copy that to the two components. One possible solution would be to put those in a separate `pub-modal.scss` file and import it as `scoped` in the components like so: `<style src="./pub-modal.scss" scoped>`. We tested that and it worked. But because it's only 4 selectors and very little rules, I think for now we'll just copy it in 2 places if I remember correctly. (@simon-debruijn can you maybe confirm this last part?)

Other things that came up:
- Listeners: As with props we're not going to emit all bootstrap events, just those we're interested in. This prevents breaking changes if a bootstrap event name changes for example, then we only have to fix it in one place.
- Constants for e.g. variants on other components like `pub-button`: We briefly discussed the upsides and downsides of this approach with constants. While there are some good advantages like having a clear list of possible values and only having to change the value in one place if it would ever change in Bootstrap, we felt it was very cumbersome to have to import the object with the constants, then create a computed value for it so we can access it in the template, and then finally use it in the template. Because of this we decided to go with "hardcoded" values in the template after all (so `variant="danger"` instead of `variant="PubButtonVariant.DANGER"`. We think we can achieve the same advantages as the constants by a) not passing the variant value directly to the bootstrap component but having a mapping in-between in our custom component, to maintain compatibility when a value changes, and b) by having a documented list of values in Storybook (has more or less the same discoverability as seeing the constants in Storybook)

I'll also document these conclusions in the docs in Storybook later on.

Reply:
I'll implement the `<style src="./pub-modal.scss" scoped>` for pub-modal now